rfc/rfc-0000-add-template-field-to-runnable.md

```diff
@@ -0,0 +1,195 @@
+          volumeClaimTemplate:
+            spec:
+              accessModes:
+                - ReadWriteOnce
+              resources:
+                requests:
+                  storage: 256Mi
+  outputs:
+    url: spec.params[?(@.name=="repository")].value
+    revision: spec.params[?(@.name=="revision")].value
+```
+
+The target persona is Cartographer maintainers; the main driver of this is to enable app operators to create templates
+that leverage Runnable without needing to submit/manage ClusterRunTemplates.
+See [this RFC](https://github.com/vmware-tanzu/cartographer/pull/705) for discussion of that use case.
+
+# How it Works
+
+[how-it-works]: #how-it-works
+
+The object defined in the Runnable is submitted to the cluster. The runnable exposes the output fields defined in
+the runnable's own status. Care should be taken to garbage collect objects submitted by a runnable if the GVK or
+name of the object is changed.
+
+# Migration
+
+[migration]: #migration
+
+See [spec-changes].
+
+This RFC would require a version bump for Runnable. There is no technical reason that Runnables with valid
+references to ClusterRunTemplate could not be translated into Runnables with template fields embedded.
+
+The [alternatives] section describes keeping `runTemplateRef` alongside `template`. This option presents a purely
+additive change.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+This runs a bit counter to the design of Cartographer, which has template objects that pair with an owner that
+provides values to the template. These template objects are meant to be reuseable. In this case they are not.
+
+This could be seen as the tail wagging the dog. Rather than simplifying the direct use case of Runnable, this makes
+the direct use case worse, in pursuit of slightly smoothing out the automation of another feature (supply chain
+templates automatically creating Runnables).
+
+# Alternatives
+
+[alternatives]: #alternatives
+
+Rather than removing ClusterRunTemplates altogether, the `template` field option could exist alongside the the
+`runTemplateRef` field. This has the advantage of preserving the owner-template relationship described in the
+[drawbacks] section above. This does lead to its own complications:
+
+- Runnable user tutorials becomes more complicated. We have to explain the choice before users. Either 1) use
+  ClusterRunTemplate and Runnable with templateRef and inputs; or 2) use Runnable with template and outputs. Failing
+  that we'd at least have to tell them to ignore the template and output fields they see in the Runnable CRD.
+- The logic of the Carto's Runnable controller becomes more complicated. The controller has to handle these two
+  different ways of specifying a Runnable.
+
+# Prior Art
+
+[prior-art]: #prior-art
+
+# Unresolved Questions
+
+[unresolved-questions]: #unresolved-questions
+
+Is the field name `template` even appropriate? Nothing is being templated, an object is completely defined in the
```

Comment:
what about `runnable.spec.selector`? even if there's no templating to be done for what used to be the `inputs` field, I imagine there'd still be the need of templating properties of the selected object? or is part of the proposal to get rid of it?